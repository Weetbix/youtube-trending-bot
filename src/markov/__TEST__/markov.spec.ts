import {
    ControlTokens,
    createDictionary,
    groupTokens,
    tokeniseSingleMessage,
} from '../markov';

describe('tokeniseMessage', () => {
    it('should handle a basic sentence', () => {
        expect(tokeniseSingleMessage('this is easy')).toEqual([
            'this',
            'is',
            'easy',
        ]);
    });
});

describe('groupTokens', () => {
    it('should not return anything when there arent enough tokens', () => {
        expect(groupTokens(['cat'], 2)).toEqual([]);
        expect(groupTokens([], 2)).toEqual([]);
    });

    it('should return correct grouping when the length matches the chain length', () => {
        expect(groupTokens(['cat'], 1)).toEqual([['cat', ControlTokens.END]]);
        expect(groupTokens(['cat', 'dog'], 2)).toEqual([
            ['cat', 'dog', ControlTokens.END],
        ]);
    });

    it('should group tokens into lengths of L + 1', () => {
        expect(groupTokens(['cat', 'dog', 'horse', 'apple'], 2)).toEqual([
            ['cat', 'dog', 'horse'],
            ['dog', 'horse', 'apple'],
            ['horse', 'apple', ControlTokens.END],
        ]);
    });
});

describe('createDictionary', () => {
    it('shoud create the expected dictionaries', () => {
        const input = 'cats and dogs';
        const tokens = tokeniseSingleMessage(input);
        const groups = groupTokens(tokens, 2);

        expect(createDictionary(groups, 2)).toMatchSnapshot();
    });

    it('shoud create the expected dictionaries when there are more than one occurance of the same word', () => {
        const input = 'cats and cats and dogs';
        const tokens = tokeniseSingleMessage(input);
        const groups = groupTokens(tokens, 2);

        expect(createDictionary(groups, 2)).toMatchSnapshot();
    });
});
