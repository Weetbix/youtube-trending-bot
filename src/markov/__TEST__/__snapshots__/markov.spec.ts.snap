// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createDictionary shoud create the expected dictionaries 1`] = `
Object {
  "and dogs": Array [
    0,
  ],
  "cats and": Array [
    "dogs",
  ],
}
`;

exports[`createDictionary shoud create the expected dictionaries when there are more than one occurance of the same word 1`] = `
Object {
  "and cats": Array [
    "and",
  ],
  "and dogs": Array [
    0,
  ],
  "cats and": Array [
    "cats",
    "dogs",
  ],
}
`;

exports[`createDictionaryFromInput should create the expected dictionary 1`] = `
Object {
  "a b": Array [
    "c",
    "c",
    "c",
  ],
  "b c": Array [
    "d!",
    "d",
    "d",
  ],
  "c d": Array [
    0,
    "d",
  ],
  "c d!": Array [
    0,
  ],
  "d d": Array [
    0,
  ],
}
`;
